### PR TITLE
Use named types for the fields of a ServerKey struct

### PR DESCRIPTION
**Pull request by NegativeMjark**

### CODE REVIEW

File: keys.go [hunk 1]
```diff
@@ -38,20 +38,35 @@ type ServerKeys struct {
...
+
+// A VerifyKey is a ed25519 public key for a server.
+type VerifyKey struct {
+	Key Base64String `json:"key"` // The public key.
+}
+
+// An OldVerifyKey is an old ed25519 public key that is no longer valid.
+type OldVerifyKey struct {
+	VerifyKey
+	ExpiredTS Timestamp `json:"expired_ts"` // When this key stopped being valid for event signing.
```
Comment (Kegsay) [inline keys.go]: Units?

File: keys.go [hunk 2]
```diff
@@ -38,20 +38,35 @@ type ServerKeys struct {
...
-		Key       Base64String `json:"key"`        // The public key.
-		ExpiredTS Timestamp    `json:"expired_ts"` // When this key stopped being valid for event signing.
-	} `json:"old_verify_keys"`
+	// The name of the server
+	ServerName string `json:"server_name"`
+	// List of SHA256 fingerprints of X509 certificates used by this server.
+	TLSFingerprints []TLSFingerprint `json:"tls_fingerprints"`
+	// The current signing keys in use on this server.
+	// These are valid while this response is valid.
+	VerifyKeys map[string]VerifyKey `json:"verify_keys"`
```
Comment (Kegsay) [inline keys.go]: What is `string`?

File: keys.go [hunk 3]
```diff
@@ -38,20 +38,35 @@ type ServerKeys struct {
...
+	// When this result is valid until in milliseconds.
+	ValidUntilTS Timestamp `json:"valid_until_ts"`
+	// Old keys that are now only valid for checking historic events.
+	OldVerifyKeys map[string]OldVerifyKey `json:"old_verify_keys"`
```
Comment (Kegsay) [inline keys.go]: What is `string`?

**Comment by Kegsay** [comment]
Otherwise LGTM

File: keys.go (hunk 1 above)
Comment (NegativeMjark) [inline keys.go]: Fixed 999dd61

File: keys.go (hunk 2 above)
Comment (NegativeMjark) [inline keys.go]: Commented 999dd61

File: keys.go (hunk 3 above)
Comment (NegativeMjark) [inline keys.go]: Commented 999dd61

**Comment by Kegsay** [comment]
LGTM

**Pull request merged**